### PR TITLE
Prevent variable parsing errors from masking validation errors

### PR DESCRIPTION
**Pull request by Shane32**
Patches #1831

### CODE REVIEW

**Comment by Shane32** [comment]
With this change, `BuildExecutionContext` sets `Variables` to `null` when `inputs` is `null`.  Previously, when `inputs` was `null`, `Variables` would be initialized as usual, assuming `null` for each individual variable.  To retain this behavior in that scenario, `Inputs.Empty` will be passed to `BuildExecutionContext` rather than `null`.

**Comment by Shane32** [comment]
For anyone that does not utilize `AfterValidationAsync`, or for people that do not access `ExecutionContext.Variables` from within `AfterValidationAsync`, this PR will entirely fix the bug described in #1831 .

However, if a user attempts to access `ExecutionContext.Variables` from within `AfterValidationAsync` when there is a problem parsing the passed variables, behavior will revert to the former behavior with the only side effect that some code within `AfterValidationAsync` has already executed (up to the point of the `NullReferenceException`).

**Comment by Shane32** [comment]
We can issue a more comprehensive change for 4.0, such as using `ValidationContext` instead of `ExecutionContext` within `AfterValidationAsync`.

File: src/GraphQL/Execution/DocumentExecuter.cs [hunk 1]
```diff
@@ -107,20 +107,70 @@ public async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options)
...
+                        options.UserContext,
+                        options.CancellationToken,
+                        metrics,
+                        options.Listeners,
+                        options.ThrowOnUnhandledException,
+                        options.UnhandledExceptionDelegate,
+                        options.MaxParallelExecutionCount,
+                        options.RequestServices);
+                }
+                catch
```
Comment (sungam3r) [inline src/GraphQL/Execution/DocumentExecuter.cs:127]: Catch should work with more specific exception.

**Comment by sungam3r** [comment]
> with the only side effect that some code within AfterValidationAsync has already executed (up to the point of the NullReferenceException).

Yes, there is some potential for side effects for "specially prepared" queries.

File: src/GraphQL/Execution/DocumentExecuter.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL/Execution/DocumentExecuter.cs:127]: Should be fine.

```suggestion
                catch (InvalidVariableError)
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**